### DISCUSSION TITLE
WindTerm 2.6.0 Prerelease 1

### DISCUSSION THREAD

**Discussion by kingToolbox** [kingToolbox/WindTerm#1093] [General]
# Spotlight

## Sync Terminal Folder
- [Intro Video: Sync Terminal Folder](https://kingtoolbox.github.io/2022/08/21/sync_terminal_folder/)

#### Sync Terminal Folder supports syncing the current working directory between filer and terminal. When you change directories in the terminal, filer should follow automatically and optionally, vice versa, the working directory of the filer can be quickly copied to the terminal.

`Sync Terminal Folder` supports:

- Copy directory path from filer to terminal.
- Manually sync terminal folder.
- Automatically sync terminal folder.

**Note that not all local or remote systems support Sync Terminal Folder.**

## Sysmon
- [Intro Video: Sysmon](https://kingtoolbox.github.io/2022/08/21/sysmon/)

#### Sysmon allows you to monitor the usage of system resources such as the time, uptime, CPU, memory, network, etc. of the local machine or remote server.

`Sysmon` supports:

- Linux (SSH, Shell)
- macOS (SSH, Shell)
- Cmd
- PowerShell
- Windows Subsystem for Linux

# Usage:
### Windows

|   Version | Architecture |  Operating Systems |
| --------- | -------------- | --------------------- |
| WindTerm_2.6.0_Prerelease_1_Windows_Portable_x86_64 | x86_64 | Windows 7, Windows 8, Windows 10, Windows 11 |

#### Versions before Version 2.4.0

1. Start `WindTerm` and select the `profiles directory` and **quit**.
2. Replace the new `.wind/profiles` folder with the old `profiles` folder.
3. Restart `WindTerm` and all the sessions and configuration will be restored.

#### Versions after Version 2.4.0

1. Start `WindTerm` and select the `profiles directory` containing the old `.wind` folder.
2. It is recommended to choose the `Home Directory` to facilitate subsequent upgrades.

If the `.wind` folder needs to be moved to another system:
1. Quit `WindTerm`.
4. Replace the new `.wind` folder with the old `.wind` folder.
5. Restart `WindTerm` and all the sessions and configuration will be restored.

The **.wind** folder is cross-platform. 

# Known Issues:
- [ZMODEM] **Only support file names contains ASCII characters in Windows, otherwise the file name of the transfered file will be garbled after transmission, but it does not affect the correctness of the file content.** Linux and MacOS version does not have this problem.

# Changes in Version 2.6.0 Prerelease 1 (2022-8-21):

# Translation:
- [GERMAN] Update the language file, contributed by **[kvnklk](https://github.com/kvnklk)**. #1031 

# Features:
- [SESSION] Supports `Sysmon` for easy monitoring of system resources usage, such as `CPU`, `memory`, `network`, etc. #65 #163 #488 #671 #803 #1077
- [FILER] Supports `Sync Terminal Folder`. #132 #193 #304 #331 #450 #814 #823 #856 #937 #1047

# Improvements:
- [GUI] Improve the display size of the dialog to better fit the monitor. #942 
- [GUI] The `autocomplete list` better supports for high DPI displays. #1021 
- [GUI] When near the bottom of the screen, the `autocomplete list` is displayed above the line of text to show more items.
- [GUI] Whether or not to beep is customizable when a message is received.

# Bugs:
- [APP] The application sometimes crashes when closing session tabs quickly. #922 
- [FILER] Unable to open a folder with the `\` character in the name.

# Intro Videos
- [Sync Terminal Folder](https://kingtoolbox.github.io/2022/08/21/sync_terminal_folder/)
- [Sysmon](https://kingtoolbox.github.io/2022/08/21/sysmon/)

<hr /><em>This discussion was created from the release <a href='https://github.com/kingToolbox/WindTerm/releases/tag/2.6-prerelease'>WindTerm 2.6.0 Prerelease 1</a>.</em>

**Comment by SadarSSI** [comment]
I have now been using Windterm daily for over a year. Unfortunately I have to go back to superputty for use SSH agent forwarding [(#77)](https://github.com/kingToolbox/WindTerm/issues/77)

Can you let us know when this feature will be supported ?

**Reply by hisune** [reply]
Sysmon not support jumpserver

![image](https://user-images.githubusercontent.com/7941669/186063859-14986696-7d8b-4d8e-8013-0a641539f500.png)

**Reply by hisune** [reply]
And `Sync Terminal Folder` not support jumpserver too. Can there be a way to support it in the future?

**Reply by JackloveJam** [reply]
我遇到的是同一局域网的开发机，我的电脑使用这个工具获取不到它的信息。但是我在开发机（VM）的宿主机使用这个工具就能看到。

**Comment by SadarSSI** [comment]
Hello, 
Thanks again for your development!

Could you consider an evolution to specify an "edit" option and specifiy default editor (for windows) when using the filer? #1104 

The filer offers an "Open" option which corresponds to the Open action referenced in Windows. This gives a surprising result if, for example, you perform open on a [*].cmd file thinking of making an edit! ;-//

Could you consider adding an "edit" option and letting the user set a default editor in the settings. Like what is done with WinSCP for example.

Indeed, an open on a file without an extension (see screenshot below) will be confusing since there is no action referenced in Windows for this type of file.

![Windterm_Filer](https://user-images.githubusercontent.com/14940543/186359836-484d3b29-69d9-46bc-9c26-69bc71d11b0c.PNG)

**Reply by roke-julian-lockhart** [reply]
Shouldn't this be a regular issue? I imagine it'll be lost here.

**Reply by ThowZzy** [reply]
![image](https://github.com/kingToolbox/WindTerm/assets/61882536/fd92b0fb-dfb5-448f-8ca2-9a6745ce9275)

& navigate to notepad++.exe

**Comment by SadarSSI** [comment]
WindTerm preview 2.6: Command history completion on linux #1147

In practice, I don't find completion under Linux to be practical / sensible. Indeed, we have under linux the history of the commands which is already managed. Propose a new completion mode why not, but let the user choose to activate it or not (...)

I reverted to 2.5.x because the command history overlay quickly becomes visually uncomfortable

**Comment by hisune** [comment]
The 2.6.0 Prerelease1 version has a certain probability of crashing, but it is impossible to confirm what action caused it. I have encountered it several times during use. I remember that sometimes it crashed when I switched the session tab.

**Comment by gxon2019** [comment]
![image](https://user-images.githubusercontent.com/53204146/196067513-1c49e4b4-036d-4bdd-98b0-effd884cbe99.png)
![image](https://user-images.githubusercontent.com/53204146/196067514-55a2020a-15fd-4c39-996d-97c45ba8a579.png)
Unable to log in to an old switching device, such as Cisco 3560, no matter how I enter the password, I will be prompted with "Authentication failed, please connect again". I can log in using other tools, such as Xshell.

**Reply by RuralHunter** [reply]
In session setting -> ssh -> authentication, uncheck other options except the password authentication

**Comment by n0rbury** [comment]
I would like to have an option to always hide the address bars.

**Reply by n0rbury** [reply]
Also, it makes sense to make the windows title bar black in dark mode.

**Reply by roke-julian-lockhart** [reply]
https://learn.microsoft.com/en-us/windows/apps/desktop/modernize/apply-windows-themes#know-when-dark-mode-is-enabled

**Comment by clownfive** [comment]
WindTerm for mac ，Please。

**Reply by n0rbury** [reply]
Actually, the [2.5.0](https://github.com/kingToolbox/WindTerm/releases/tag/2.5.0) is already available for mac and linux. You can try that out. I believe the 2.6.0 release will also include mac and linux versions.

**Reply by yurenchen000** [reply]
linux users also waiting for **pre-release**

**Comment by mayabinliu** [comment]
2.6.0 release for mac  please

**Reply by yurenchen000** [reply]
linux user also looking forward the prerelease tar.gz

**Reply by edrianolima** [reply]
im too wait for 2.6.0 for mac

**Comment by shaojs321** [comment]
最爱的两个功能在同一个版本中出现，感谢作者，加油！

**Comment by liuxiaomingskm** [comment]
Did you give up upgrading this tool after 5 months passed by？

**Reply by roke-julian-lockhart** [reply]
No - https://github.com/kingToolbox/WindTerm/issues/1596 (read the *whole* thing)

**Reply by liuxiaomingskm** [reply]
scared to me! Then I realized it is just a April's joke!

**Comment by ghost** [comment]
这个版本使用SSH协议登录交换机时界面会卡住直到失去连接，但是登录服务器不会。目前感觉可能跟登录扫码文件有关系，2.5版本可以正常使用，但是没有文件跟踪。

**Comment by danaokezi** [comment]
1、建议增加可以修改和自主添加命令提示功能，然后还可以开发出不同领域的常用命令包，满足不同工作领域的代码提示需求。
2、提供chatGPT接口，以达到ai提示。此功能在warp已经实现，并在mac圈逐渐火起来，但warp是收费并且目前不支持windows和linux。此功能只需要提供配置接入chatGPT等智能接口即可，增强用户的扩展性强，并且可以增加产品竞争力。

**Comment by danaokezi** [comment]
1、I suggest adding the ability to modify and independently add command prompt functionality, and developing commonly used command packages in different fields to meet the code prompt needs of different work domains.

2、Provide a ChatGPT interface to achieve AI prompts. This feature has already been implemented in Warp and is gradually gaining popularity in the Mac community, but Warp is a paid service and currently does not support Windows and Linux. This feature only needs to provide configuration access to intelligent interfaces such as ChatGPT, which enhances user scalability and increases product competitiveness.

**Reply by alterdaemon** [reply]
integrating ChatGPT into terminal/ssh app is imo a terrible idea and simple road into the overbloat.
there are many such tools already available with CLI interface. e.g. look at [yai](https://github.com/ekkinox/yai).

I hope that the author of this great software will keep on being reasonable (that means conservative) regarding it's features expansion.
@kingToolbox

**Comment by lufengQwQ** [comment]
Hope have mac arm version.😘

**Reply by milkday** [reply]
me too

**Comment by MaxTarasenko** [comment]
Waiting for Mac version
How can I build a Mac version from your sources?

**Reply by roke-julian-lockhart** [reply]
@MaxTarasenko, doesn't the ReadMe state how to compile it?

**Reply by MaxTarasenko** [reply]
No > Doesn't the ReadMe state how to compile it?

**Comment by roke-julian-lockhart** [comment]
Like https://github.com/kingToolbox/WindTerm/discussions/1093#discussioncomment-4998129, please post a compiled version for Linux.

**Comment by SunPan92** [comment]
When using sftp to upload files, if you enter commands on the terminal interface at this time, there is a certain chance that the files will become invalid.
At that time, I was uploading a springboot program file to the server, and input instructions in the terminal. After the upload was complete, I used the ` java -jar`  command to start the program, and an error message indicated that this was not a jar file.
Then, I re-uploaded the file again. At this time, no command was entered. After the file upload was completed, I could use the command to start the program.

**Comment by zorrohahaha** [comment]
When the 2.6 version will be finally released?

**Comment by elicec** [comment]
CPU load high

**Comment by iHuifeideyu** [comment]
win11中,在两个不同分辨率显示屏中,分辨率高的显示屏中WindTerm 2.6.0 Prerelease 1界面会变大,如果在属性中设置高DPI界面字体会变模糊.而WindTerm 2.5.0在两个显示屏中大小一样,字体不会变模糊.
In Windows 11, on two different resolutions, the interface of WindTerm 2.6.0 Prerelease 1 on the higher-resolution display becomes larger, and the font becomes blurry if the high-DPI interface is set in the properties. However, WindTerm 2.5.0 has the same size on both displays, and the font does not become blurry.

**Comment by Eif-Hui** [comment]
win11中，会话设置，"保存"按钮置灰，无法点击

**Comment by anhthuongtran502** [comment]
Hi support,
Could you please tell me how to disable the "OneKey" option when doing login to a destination?
I dont want to see the "OneKey" box there, I just want the Account is enabled by default.
Note: I am using the WindTerm_2.6.0 portable
![image](https://github.com/kingToolbox/WindTerm/assets/125422297/b58488e5-ac3a-4672-abfc-277945e78194)

**Comment by ablewolf** [comment]
mac The mac version cannot be opened.
I tried inter chip computer and M1 chip computer but couldn't open it.

**Comment by ruizhang3** [comment]
mac computers cannot open software m2

**Comment by amaoo** [comment]
@kingToolbox 

Currently, many servers have upgraded their VIM version to V9, but there is a problem that ESC[?4m appears when editing a file, although it can be fixed temporarily by the following ways, but it is always troublesome.

Write these into .vimrc:

```
set keyprotocol=
let &term=&term
```

**Comment by Calcifer97** [comment]
@kingToolbox linux版本什么时候发出呢

**Comment by larkzhang** [comment]
When will macos be released

**Comment by 0zyt** [comment]
release10版本好像有点bug，在远程的Linux中切换目录左边的文件管理器不会跟着变目录了

**Comment by nyakang** [comment]
Syncing terminal folders does not work in Prerelease 10

**Comment by jerry2wang** [comment]
After logging into the Linux system on the terminal and opening a window using tmux, the shortcut key Ctrl+l for clearing the screen does not work.
